src/bridge/BitcoinTx.sol

```diff
@@ -280,4 +293,32 @@ library BitcoinTx {
 
         revert("No output found for scriptPubKey");
     }
+
+    function checkOutboundTxInputsMatchesUtxo(bytes memory TxInputVector, BitcoinTx.UTXO calldata utxo)
+        internal
+        pure
+        returns (bytes32 outpointTxHash, uint32 outpointIndex)
+    {
+        // To determine the total number of Bitcoin transaction inputs,
+        // we need to parse the compactSize uint (VarInt) the input vector is
+        // prepended by. That compactSize uint encodes the number of vector
+        // elements using the format presented in:
+        // https://developer.bitcoin.org/reference/transactions.html#compactsize-unsigned-integers
+        // We don't need asserting the compactSize uint is parseable since it
+        // was already checked during `validateVin` validation.
+        // See `BitcoinTx.inputVector` docs for more details.
+        (, uint256 inputsCount) = TxInputVector.parseVarInt();
+
+        for (uint256 inputId = 0; inputId < inputsCount; inputId++) {
+            bytes memory input = TxInputVector.extractInputAtIndex(inputId);
```

Comment:
Since this would elicit quadratic complexity, better to follow what we do here: https://github.com/bob-collective/bitcoin-spv/blob/98264ce672c60509e7ac7fbbfd70f0ad0d31d292/src/BTCUtils.sol#L279C17-L279C17

Reply:
updated `ensureTxInputSpendsUtxo` to create slice inputs from TxInputvector, and checks if it matches utxo tx hash